crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -190,31 +198,71 @@ pub enum RoomListLoadingState {
 
 type BoxedFilterFn = Box<dyn Fn(&RoomListEntry) -> bool + Send + Sync>;
 
-/// Dynamic filter for the [`RoomList`] entries.
+/// Controller for the [`RoomList`] dynamic entries.
 ///
-/// To get one value of this type, use [`RoomList::entries_with_dynamic_filter`]
-pub struct DynamicRoomListFilter {
-    inner: Arc<AsyncCell<BoxedFilterFn>>,
+/// To get one value of this type, use
+/// [`RoomList::entries_with_dynamic_adapters`]
+pub struct RoomListDynamicEntriesController {
+    filter: Arc<AsyncCell<BoxedFilterFn>>,
+    page_size: usize,
```

Comment:
This could be a u32, right? Would avoid a `try_into`, and u32::max_value() entries per page should be enough for everyone...

Reply:
You're right! It removes the conversion from `u32` to `usize` in the FFI binding too.

Reply:
Actually… hmm… I need **more** conversions from `u32` to `usize` if I change `page_size` to `u32`, because `limit` is a `Stream<Item = usize>`. It's better as is.